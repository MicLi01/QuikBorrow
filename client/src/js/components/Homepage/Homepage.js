import React, { useState } from "react";
import styled from 'styled-components';
import { Typography, Button, ButtonGroup, Dialog, DialogTitle, DialogContent, DialogActions, TextField, FormControl, FormHelperText, InputAdornment, MenuItem } from "@material-ui/core";
import teal from '@material-ui/core/colors/teal'
import ChatIcon from '@material-ui/icons/Chat';
import { VerifiedUserTwoTone, ThumbsUpDownTwoTone, AddCircle } from "@material-ui/icons";
import Dropzone from 'react-dropzone'
import ImageIcon from '@material-ui/icons/Image'
import AddItemModal from '../AddItemModal/AddItemModal'

const Background = styled.div`
	background: #009688;
	height: 500px;
	background-position: center;
	background-repeat: no-repeat;
	background-size: cover;
	margin-bottom: -350px;
`

const Content = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	h4 {
		margin-bottom: 20px;
	}
	p {
		margin-bottom: 20px;
	}
	margin-bottom: 210px;
`

const Section = styled.section`
	padding: 0 50px;
	margin: 0 auto;
	display: flex;
	justify-content: center;
	flex-direction: column;
	max-width: 1100px;
`

const Cards = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	margin-top: 40px;
`

const Card = styled.div`
	display: flex;
	text-align: left;
	width: 100%;
	justify-content: space-between;
	margin-bottom: 20px;
	> div {
		padding-right: 20px;
	}
`

const Heading = styled(Typography)`
	color: ${teal[800]};
`

function Homepage() {
	const [ openModal, setOpenModal ] = useState(false)

	return (
		<div>
			<Background />
			<AddItemModal open={openModal} onClose={e => setOpenModal(false)} />
			<Content>
				<Typography variant='h4' component='h4' style={{color: teal[50]}}>
					Need something?
				</Typography>
				<Typography align='center' style={{color: teal[50]}} variant='body1'>
					With Quikborrow, you can do whatever you want with no reprucussions.<br/>What do you want to do?
				</Typography>
				<ButtonGroup>
					<Button size='large' variant='contained' onClick={e => setOpenModal(true)}>
						I want to lend
					</Button>
					<Button size='large' variant='contained'>
						I want to borrow
					</Button>
				</ButtonGroup>
			</Content>
			<Section>
				<Heading variant='h4' component='h4' gutterBottom>
					By the students. For the students.
				</Heading>
				<Typography variant='h6' component='p' paragraph gutterBottom>
					Too long have we been chained by our corporate overlords such as McGraw Hill and Pearson. Today, you have the chance to break free with QuikBorrow. A service so quik you can't even "c" it. *<i>pause for applause</i>*
				</Typography>
				<Cards>
					<Card>
						<div>
							<Heading variant='h5' component='h5' gutterBottom>
								Securely Chat
							</Heading>
							<Typography variant='body1' component='p'>
								The NSA is always watching our every move. On our platform however, they won't see a thing. You are free to trade your musty second-hand textbooks with those suspicious liquid stains from one student to another without any prying eyes.
							</Typography>
						</div>
						<ChatIcon style={{fontSize: 150, color: teal[500]}} />
					</Card>
					<Card>
						<div>
							<Heading variant='h5' component='h5' gutterBottom>
								Trusted Users
							</Heading>
							<Typography variant='body1' component='p'>
								Students should feel confident that the stuff they're borrowing are not bugged by corporate spies. Users will rate each other on their trustworthiness, and their ratings will be affected for everyone to see. User reports can also be made to our team that will either confirm or deny corporate spies who intends to break our service.
							</Typography>
						</div>
						<ThumbsUpDownTwoTone style={{fontSize: 150, color: teal[500]}} />
					</Card>
					<Card>
						<div>
							<Heading variant='h5' component='h5' gutterBottom>
								Students Only
							</Heading>
							<Typography variant='body1' component='p'>
								Corporate spies are the bane of our existence, and we use a wide range of state-of-the-art tech tools to keep them out. All lenders and buyers must validate their account with student emails to verify that they are attending a university. Once they're no longer a student, they can no longer lend or borrow items.
							</Typography>
						</div>
						<VerifiedUserTwoTone style={{fontSize: 150, color: teal[500]}} />
					</Card>
				</Cards>
			</Section>
		</div>
	);
}
export default Homepage;
