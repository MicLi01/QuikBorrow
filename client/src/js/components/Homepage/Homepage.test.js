import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import Homepage from './Homepage';
const mockStore = configureMockStore()

describe('Homepage', () => {
	test('renders', () => {
		const store = mockStore({
			auth: {
				isAuthenticated: false
			}
		})
		const component = renderer.create(
			<Provider store={store}>
				<MemoryRouter>
					<Homepage />
				</MemoryRouter>
			</Provider>
		).toJSON();
		
		expect(component).toMatchSnapshot();
	});
});