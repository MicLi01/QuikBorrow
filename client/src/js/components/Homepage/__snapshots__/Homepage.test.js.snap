// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders 1`] = `
<div>
  <div
    className="sc-ifAKCX dPYUdS"
  />
  <div
    className="sc-EHOje djFjdi"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
      style={
        Object {
          "color": "#e0f2f1",
        }
      }
    >
      Need something?
    </h4>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      style={
        Object {
          "color": "#e0f2f1",
        }
      }
    >
      With Quikborrow, you can do whatever you want with no reprucussions.
      <br />
      What do you want to do?
    </p>
    <div
      className="MuiButtonGroup-root"
      role="group"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          I want to lend
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          I want to borrow
        </span>
      </button>
    </div>
  </div>
  <section
    className="sc-bZQynM jaZHyE"
  >
    <h4
      className="MuiTypography-root sc-dnqmqq iKcNkm MuiTypography-h4 MuiTypography-gutterBottom"
    >
      By the students. For the students.
    </h4>
    <p
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-paragraph"
    >
      Too long have we been chained by our corporate overlords such as McGraw Hill and Pearson. Today, you have the chance to break free with QuikBorrow. A service so quik you can't even "c" it. *
      <i>
        pause for applause
      </i>
      *
    </p>
    <div
      className="sc-gzVnrw chfBbm"
    >
      <div
        className="sc-htoDjs KzKsL"
      >
        <div>
          <h5
            className="MuiTypography-root sc-dnqmqq iKcNkm MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Securely Chat
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            The NSA is always watching our every move. On our platform however, they won't see a thing. You are free to trade your musty second-hand textbooks with those suspicious liquid stains from one student to another without any prying eyes.
          </p>
        </div>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "color": "#009688",
              "fontSize": 150,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 9h12v2H6V9zm8 5H6v-2h8v2zm4-6H6V6h12v2z"
          />
        </svg>
      </div>
      <div
        className="sc-htoDjs KzKsL"
      >
        <div>
          <h5
            className="MuiTypography-root sc-dnqmqq iKcNkm MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Trusted Users
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Students should feel confident that the stuff they're borrowing are not bugged by corporate spies. Users will rate each other on their trustworthiness, and their ratings will be affected for everyone to see. User reports can also be made to our team that will either confirm or deny corporate spies who intends to break our service.
          </p>
        </div>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "color": "#009688",
              "fontSize": 150,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 6c0-.55-.45-1-1-1H5.82l.66-3.18.02-.23c0-.31-.13-.59-.33-.8L5.38 0 .44 4.94C.17 5.21 0 5.59 0 6v6.5c0 .83.67 1.5 1.5 1.5h6.75c.62 0 1.15-.38 1.38-.91l2.26-5.29c.07-.17.11-.36.11-.55V6zm-2 1.13L7.92 12H2V6.21l1.93-1.93L3.36 7H10v.13zM22.5 10h-6.75c-.62 0-1.15.38-1.38.91l-2.26 5.29c-.07.17-.11.36-.11.55V18c0 .55.45 1 1 1h5.18l-.66 3.18-.02.24c0 .31.13.59.33.8l.79.78 4.94-4.94c.27-.27.44-.65.44-1.06v-6.5c0-.83-.67-1.5-1.5-1.5zm-.5 7.79l-1.93 1.93.57-2.72H14v-.13L16.08 12H22v5.79z"
          />
          <path
            d="M3.93 4.28L2 6.21V12h5.92L10 7.13V7H3.36zM14 16.87V17h6.64l-.57 2.72L22 17.79V12h-5.92z"
            opacity=".3"
          />
        </svg>
      </div>
      <div
        className="sc-htoDjs KzKsL"
      >
        <div>
          <h5
            className="MuiTypography-root sc-dnqmqq iKcNkm MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Students Only
          </h5>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Corporate spies are the bane of our existence, and we use a wide range of state-of-the-art tech tools to keep them out. All lenders and buyers must validate their account with student emails to verify that they are attending a university. Once they're no longer a student, they can no longer lend or borrow items.
          </p>
        </div>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "color": "#009688",
              "fontSize": 150,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm7 10c0 4.52-2.98 8.69-7 9.93-4.02-1.24-7-5.41-7-9.93V6.3l7-3.11 7 3.11V11zm-11.59.59L6 13l4 4 8-8-1.41-1.42L10 14.17z"
          />
          <path
            d="M5 6.3V11c0 4.52 2.98 8.69 7 9.93 4.02-1.23 7-5.41 7-9.93V6.3l-7-3.11L5 6.3zM18 9l-8 8-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9z"
            opacity=".3"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;
